### PR TITLE
Renamed lr_dict to lr_scheduler_config

### LINKED ISSUE #9300: Misleading documentation for OneCycleLR, LambdaLR
Version 1.4.4

Per https://github.com/PyTorchLightning/pytorch-lightning/issues/4929#issuecomment-739028197 OneCycleLR, LambdaLR require configuration `"interval": "step"` which should be set as follows:

```
  'lr_scheduler': {
          'scheduler': scheduler, 
          'interval': 'step'
  }
```

But https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers implies it should be set as follows:

```
  'lr_scheduler': scheduler, 
  'interval': 'step'
```

I can confirm the first approach works. The documentation in the second link should be reviewed. I suspect it is incorrect.

### PR DESCRIPTION
**Pull request by cowwoc**
## What does this PR do?

Fixes #9300

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Always :)

### CODE REVIEW

**Comment by cowwoc** [comment]
Any idea why the benchmark CI is failing. Is this an intermittent error?

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by cowwoc** [comment]
@tchaton Thanks! Is there a way to retry the benchmarking step?

**Comment by tchaton** [comment]
Screenshot 2021-09-03 at 18 45 14

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
